File: exercise/day01/php/day01/src/Communication/SantaCommunicator.php
<?php

declare(strict_types = 1);

namespace Communication {
	class SantaCommunicator
	{
		private int $numberOfDaysToRest;

		public function __construct(int $numberOfDaysToRest)
		{
			$this->numberOfDaysToRest = $numberOfDaysToRest;
		}

		public function composeMessage(string $reindeerName, string $currentLocation, int $numbersOfDaysForComingBack, int $numberOfDaysBeforeChristmas): string
		{
			$daysBeforeReturn = $this->daysBeforeReturn($numbersOfDaysForComingBack, $numberOfDaysBeforeChristmas);

			return "Dear {$reindeerName}, please return from {$currentLocation} in {$daysBeforeReturn} day(s) to be ready and rest before Christmas.";
		}

		public function isOverdue(string $reindeerName, string $currentLocation, int $numbersOfDaysForComingBack, int $numberOfDaysBeforeChristmas, ILogger $logger): bool
		{
			if ($this->daysBeforeReturn($numbersOfDaysForComingBack, $numberOfDaysBeforeChristmas) <= 0) {
				$logger->log("Overdue for {$reindeerName} located {$currentLocation}.");

				return true;
			}

			return false;
		}

		private function daysBeforeReturn(int $numbersOfDaysForComingBack, int $numberOfDaysBeforeChristmas): int
		{
			return $numberOfDaysBeforeChristmas - $numbersOfDaysForComingBack - $this->numberOfDaysToRest;
		}
	}
}

File: solution/day01/src/main/java/communication/SantaCommunicator.java
package communication;

public class SantaCommunicator {
    private final Configuration configuration;

    public SantaCommunicator(Configuration configuration) {
        this.configuration = configuration;
    }

    public String composeMessage(Reindeer reindeer) {
        return "Dear " + reindeer.name() + ", please return from " + reindeer.location() +
                " in " + daysBeforeReturn(reindeer) + " day(s) to be ready and rest before Christmas.";
    }

    public boolean isOverdue(Reindeer reindeer, Logger logger) {
        if (daysBeforeReturn(reindeer) <= 0) {
            logger.log("Overdue for " + reindeer.name() + " located " + reindeer.location() + ".");
            return true;
        }
        return false;
    }

    private int daysBeforeReturn(Reindeer reindeer) {
        return configuration.numberOfDayBeforeChristmas() - reindeer.numbersOfDaysForComingBack() - configuration.numberOfDaysToRest();
    }
}